src/main/com/yetanalytics/flint/axiom/impl.cljc

```diff
@@ -0,0 +1,441 @@
+  (-format-rdf-type [sym] (name sym)))
+
+;; Defaults
+
+(extend-protocol-default p/Variable
+                         #?(:clj [java.lang.Object nil] :cljs default)
+                         (-valid-variable? [_])
+                         (-format-variable [_]))
+
+(extend-protocol-default p/BlankNode
+                         #?(:clj [java.lang.Object nil] :cljs default)
+                         (-valid-bnode? [_])
+                         (-format-bnode [_]))
+
+(extend-protocol-default p/Wildcard
+                         #?(:clj [java.lang.Object nil] :cljs default)
+                         (-valid-wildcard? [_])
+                         (-format-wildcard [_]))
+
+(extend-protocol-default p/RDFType
+                         #?(:clj [java.lang.Object nil] :cljs default)
+                         (-valid-rdf-type? [_])
+                         (-format-rdf-type [_]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Basic Literals (strings, lang maps, booleans)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(extend-protocol p/Literal
+  #?(:clj java.lang.String
+     :cljs string)
+  (-valid-literal?
+    [s]
+    (val-impl/valid-string-literal? s))
+  (-format-literal
+    ([s] (str "\"" s "\"")) ; Special treatment of plain string literals
+    ([s {:keys [force-iri?] :as opts}]
+     (if force-iri?
+       (fmt-impl/format-literal s opts)
+       (p/-format-literal s))))
+  (-format-literal-strval [s] s)
+  (-format-literal-lang-tag [_] nil)
+  (-format-literal-url
+    ([s] (p/-format-literal-url s {}))
+    ([_ opts] (fmt-impl/format-xsd-iri "string" opts)))
+
+  #?(:clj clojure.lang.IPersistentMap
+     :cljs PersistentArrayMap)
+  (-valid-literal? [m]
+    (val-impl/valid-lang-map-literal? m))
+  (-format-literal
+    ([m] (fmt-impl/format-lang-map-literal m))
+    ([m _] (fmt-impl/format-lang-map-literal m)))
+  (-format-literal-strval [m]
+    (fmt-impl/format-lang-map-val m))
+  (-format-literal-lang-tag [m]
+    (fmt-impl/format-lang-map-tag m))
+  (-format-literal-url ; Always unused during formatting
+    ([n] (p/-format-literal-url n {}))
+    ([_ opts] (fmt-impl/format-rdf-iri "langString" opts)))
+
+  #?(:clj java.lang.Boolean
+     :cljs boolean)
+  (-valid-literal? [_] true)
+  (-format-literal
+    ([n] (p/-format-literal-strval n))
+    ([n opts] (fmt-impl/format-literal n opts)))
+  (-format-literal-strval [n] (.toString n))
+  (-format-literal-lang-tag [_] nil)
+  (-format-literal-url
+    ([n] (p/-format-literal-url n {}))
+    ([_ opts] (fmt-impl/format-xsd-iri "boolean" opts))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Numeric Literals
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+#?(:clj
+   (extend-protocol p/Literal
+     ;; See: "XSD data types" in
+     ;; https://jena.apache.org/documentation/notes/typed-literals.html
+     java.lang.Float
```

Comment:
@RickMoynihan Given what you've said about other protocols, I wonder what you think of extending the `Literal` protocol to each of Java's different numeric types. I did this so I can be more exact with the datatype URLs (i.e. `xsd:int`, `xsd:short`, `xsd:byte`, and `xsd:long` instead of just `xsd:integer`), but maybe we can afford to extend less types in practice?

Reply:
I personally think you should and did a similar mapping in grafter: 

https://github.com/Swirrl/grafter/blob/e921df18e60a4903cd84a7cd402591891ac4581d/src/grafter_2/rdf/protocols.cljc#L303-L377

Reply:
I notice that you have support for `clojure.lang.BigInt` so I added that (and shortened `integer` to `int` for `Integer` values) in https://github.com/yetanalytics/flint/pull/27/commits/59b2e079f3bb37a2aa2e272d193d226edebcbcf9